### PR TITLE
Bayonets and other attachment changes

### PR DESCRIPTION
**Pull request by Alliostra**
## About The Pull Request

- Adds the ability to attach the tactical knife to all guns (except for the bolt guns for obvious reasons). It increases the melee damage by 5 and the melee armor penetration by 15. However, it adds a small, constant amount of inaccuracy. the damage reduction has been removed, I did not account for the fact that a gun can only hold 3 attachments at most.

- Adds the ability to attach the gyroscopic stabilizer to guns. It decreases the recoil penalty for holding a gun by 80%, but also decreases it's AP by 40%.

- Silenced gunshots can now be heard in a radius of 4 tiles around the user instead of 8 tiles, and the Silencer now lowers the recoil of the weapon by 10%

- Due to an oversight, attaching a scope caused a small loss of accuracy, this has been fixed

- The bolt rifle now has 20 melee AP. to account for this, the melee damage of the serbian version has been lowered by 3

- Adds the makeshift bullet time generator, crafted from a gravity barrel and cable coil. When attached to a ballistic gun, it slows all projectiles down by 90%, and majorly increases the firerate of the weapons.

- gunmods that add radiation damage now properly apply the radiation

- buffs the glass widow to deal 100 rad damage and the atomik to deal 30 rad damage

- buffs the decloner to deal 150 rad damage, and makes the plasma pistol deal no more rad damage. it's toxin damage has been raised to 25 as a compensation

Radiation damage applies its value to the radiation level of someone, and that level can't be raised above 100, so two decloner bolts have no added effect, unless the target is wearing rad-proof armor. Even with a rad level of 100, it still takes you about 6 minutes to accumulate around 30 tox damage, so this is by no means OP. the biggest strength is the chance to deal a tiny bit of clone damage, anyways, so I don't think this will have a big impact on game balance

Addendum:

- The overdrive chip now gives energy guns a fullauto mode, lowers the charge consumed per shot by 85% and increases recoil by 200%. Keep in mind that it also reduces damage by a whopping 80%, so this should not be a huge balance issue. Now also lowers agony damage.

- You can now no longer attach the overshooter to the AMR and the double-barrel (for balance reasons, and those guns don't have magazines anyways, you just put the bullets into a fancy metal tube.)

-increased the fire delay and recoil penalty of the gauss coil, extra damage, AP and wall pen should not come at a low cost.

## Changelog
:cl:
tweak: you can now attach the tactical knife and the gyrostabilized grip to guns
add: added the makeshift bullet time generator, a craftable gunmod
balance: silencers are now more silent and lower the recoil of your gun somewhat
balance: nerfed the novakovic's melee damage, increased it's melee AP
fix: attaching scopes no longer worsen your gun's accuracy
fix: radiation-inducing gunmods now properly apply their rads
balance: buffed the radiation-inducing gunmods
balance: the decloner now deals significantly more rad damage
balance: the plasma pistol no longer deals rad damage, but deals more tox damage now
fix: you can no longer attach the overshooter to the AMR and the double-barreled shotgun
balance: the Tesla chip now enables an energy gun to shoot in fullauto and makes it consume much less charge per shot.
/:cl:

### CODE REVIEW

File: code/game/objects/items/weapons/tools/knives.dm [hunk 1]
```diff
@@ -88,6 +88,19 @@
 	armor_penetration = ARMOR_PEN_MODERATE
 	max_upgrades = 3
 
+/obj/item/weapon/tool/knife/tacknife/New()
+	..()
+	var/datum/component/item_upgrade/I = AddComponent(/datum/component/item_upgrade)
+	I.weapon_upgrades = list(
+		GUN_UPGRADE_BAYONET = TRUE,
+		GUN_UPGRADE_MELEEDAMAGE = 5,
```
Comment (MadmanMartian) [inline code/game/objects/items/weapons/tools/knives.dm:96]: Remember to reset these in refresh_upgrades, gun.dm

**Review by MadmanMartian — DISMISSED** [review]

File: code/modules/projectiles/guns/projectile.dm [hunk 2]
```diff
@@ -371,5 +371,5 @@
 		if(CAL_PISTOL)
 			gun_tags |= GUN_CALIBRE_35
 		//Others to be implemented when needed
-	if(max_shells)
+	if(max_shells > 1) // so the
```
Comment (mattroks101) [inline code/modules/projectiles/guns/projectile.dm:374]: so the?

Comment (Alliostra) [reply code/modules/projectiles/guns/projectile.dm:374]: forgot this existed, was supposed to make it impossible to attach the overshooter to an AMR

**Comment by Alliostra** [comment]
While doing further testing, I noticed some incosistencies- previous to the changes in this PR, you used to be able to bring the radiation level above 100 with irradiating weapons, now it's capped at the 100%.

**Review by genessee596 — APPROVED** [review]

File: code/modules/projectiles/guns/projectile.dm [hunk 3]
```diff
@@ -371,5 +371,5 @@
 		if(CAL_PISTOL)
 			gun_tags |= GUN_CALIBRE_35
 		//Others to be implemented when needed
-	if(max_shells)
+	if(max_shells > 2) // so the overshooter can't be attached to the AMR and double-barrel anymore
```
Comment (TheShown911) [inline code/modules/projectiles/guns/projectile.dm:374]: No fun.

Comment (Alliostra) [reply code/modules/projectiles/guns/projectile.dm:374]: Eris

Comment (TheShown911) [reply code/modules/projectiles/guns/projectile.dm]: Yes fun i changed my mind.

**Comment by drexample** [comment]
Fix conflicts

File: code/datums/craft/recipes/weapon.dm [hunk 4]
```diff
@@ -230,6 +230,15 @@
 		list(QUALITY_ADHESIVE, 15, 70)
 	)
 
+/datum/craft_recipe/weapon/gravcharger
+	name = "Makeshift bullet time generator "
```
Comment (TheShown911) [inline code/datums/craft/recipes/weapon.dm:234]:
```suggestion
	name = "Makeshift bullet time generator"
```

File: code/modules/projectiles/guns/mods/mods.dm [hunk 5]
```diff
@@ -134,8 +134,8 @@
 	..()
 	var/datum/component/item_upgrade/I = AddComponent(/datum/component/item_upgrade)
 	I.weapon_upgrades = list(
-		GUN_UPGRADE_DAMAGE_RADIATION = 5
-		)
+		GUN_UPGRADE_DAMAGE_RADIATION = 100
```
Comment (TheShown911) [inline code/modules/projectiles/guns/mods/mods.dm:137]: This is big.

**Review by TheShown911 — APPROVED** [review]
Approved

File: code/modules/projectiles/guns/mods/mods.dm (hunk 5 above)
Comment (Alliostra) [reply code/modules/projectiles/guns/mods/mods.dm:137]: Not as big as it appears to be, I tested it

**Pull request merged**